src/ray/raylet/node_manager.cc

```diff
@@ -1608,7 +1609,48 @@ void NodeManager::HandleRequestWorkerLease(const rpc::RequestWorkerLeaseRequest
     worker_pool_.PrestartWorkers(task_spec, request.backlog_size());
   }
 
-  cluster_task_manager_->QueueAndScheduleTask(task, reply, send_reply_callback);
+  if (!(RayConfig::instance().gcs_actor_scheduling_enabled() &&
+        task.GetTaskSpecification().IsActorCreationTask())) {
+    cluster_task_manager_->QueueAndScheduleTask(task, reply, send_reply_callback);
+    return;
+  }
+
+  auto send_reply_callback_wrapper = [this, actor_id, reply, send_reply_callback](
+                                         Status status, std::function<void()> success,
+                                         std::function<void()> failure) {
+    if (!reply->rejected()) {
+      send_reply_callback(status, success, failure);
+      return;
+    }
+
+    // If the reqiured resource and normal task resource exceed available resource,
+    // reject it.
+    ResourceSet normal_task_resources = cluster_task_manager_->CalcNormalTaskResources();
+    std::string msg = "Reject leasing as the raylet has no enough resources.";
```

Comment:
Put the msg directly in RAY_LOG(INFO).
Btw, should it be RAY_LOG(DEBUG) ??

Reply:
This is rare event in most cases, and is worth exposing to users.